src/App.tsx

```diff
@@ -1,93 +1,93 @@
+  email: string;
+  id: number;
+  name: string;
+}
 
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
+export const App = () => {
+  function useLocaleStorage<T>(
+    key: string,
+    initialValue: T,
+  ): [T, (value: T) => void] {
+    const [value, setValue] = useState<T>(() => {
+      try {
+        const item = localStorage.getItem(key);
 
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
-        />
-        <label htmlFor="toggle-all">Mark all as complete</label>
+        return item ? JSON.parse(item) : initialValue;
+      } catch {
+        return initialValue;
+      }
+    });
 
-        <ul className="todo-list" data-cy="todoList">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view" />
-              <label htmlFor="toggle-view">asdfghj</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    const save = (v: T): void => {
+      setValue(v);
+      localStorage.setItem(key, JSON.stringify(v));
+    };
 
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-completed" />
-              <label htmlFor="toggle-completed">qwertyuio</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+    return [value, save];
+  }
 
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-editing" />
-              <label htmlFor="toggle-editing">zxcvbnm</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const [userEmail, setUserEmail] = useState('');
+  const [userName, setUserName] = useState('');
+  const [errorInput, setErrorInput] = useState<ErrorInput>(ErrorInput.NONE);
+  const [secVisIn, setSecVisIn] = useState(false);
+  const [user, setUser] = useLocaleStorage<User[] | []>('user', []);
 
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view2" />
-              <label htmlFor="toggle-view2">1234567890</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
-      </section>
+  const addNewUser = (name:string, email:string) => {
```

Comment:
create `users.ts` inside api folder for these functions (in the same way as todos.ts)